lib/checks/color/color-contrast-evaluate.js

```diff
@@ -71,14 +71,16 @@ export default function colorContrastEvaluate(node, options, virtualNode) {
   if (shadowColors.length === 0) {
     contrast = getContrast(bgColor, fgColor);
   } else if (fgColor && bgColor) {
-    // Thin shadows can pass either by contrasting with the text color
-    // or when contrasting with the background.
     shadowColor = [...shadowColors, bgColor].reduce(flattenShadowColors);
-    const bgContrast = getContrast(bgColor, shadowColor);
-    const fgContrast = getContrast(shadowColor, fgColor);
-    contrast = Math.max(bgContrast, fgContrast);
-    contrastContributor =
-      bgContrast > fgContrast ? 'shadowOnBgColor' : 'fgOnShadowColor';
+    // Compare shadow, bgColor, textColor. Check passes if any is sufficient
+    const fgBgContrast = getContrast(bgColor, fgColor);
```

Comment:
This is the critical change here. Considering fgColor / bgColor contrast even if there's a thin shadow.